### PR TITLE
Issue 357: Move from fixed n with missing to flexible n

### PR DESCRIPTION
**Pull request by seabbs**
This PR expands on the ideas in #389 and from f2f conversations to try to partially resolve some issues seen in #357. Before merging it needs a benchmark and test update to check that it does this. 

I think from this implementation their are few user side downsides apart from potentially some confusion about the changing length of complex latent models.

### CODE REVIEW

**Review by SamuelBrand1 — DISMISSED** [review]
LGTM. Will be interested if this has any noticeable effect on benchmarks.

File: EpiAware/src/EpiObsModels/ObservationErrorModels/methods.jl [hunk 1]
```diff
@@ -10,15 +12,16 @@ It dispatches to the `observation_error` function to generate the observation er
...
     end
 
+    diff_t = length(y_t) - length(Y_t)
+    @assert diff_t>=0 "The observation vector must be longer than or equal to the expected observation vector"
+
     pad_Y_t = Y_t .+ 1e-6
 
-    for i in findfirst(!ismissing, Y_t):length(Y_t)
-        y_t[i] ~ observation_error(obs_model, pad_Y_t[i], priors...)
+    for i in eachindex(Y_t)
```
[the remark this answers is not in the document]
Comment (seabbs) [reply EpiAware/src/EpiObsModels/ObservationErrorModels/methods.jl:23]: as in make it vectorised when all data is present? Yes I think so and should likely speed things up?

[the remark this answers is not in the document]
Comment (seabbs) [reply EpiAware/src/EpiObsModels/ObservationErrorModels/methods.jl:23]: want to make an issue? I think I'm not quite understanding what you mean enough to do so.

**Comment by seabbs** [comment]
I've added integration benchmarks (including from #357 which now run without check warnings being thrown (as far as I can see). I haven't added new unit tests as I felt like the functionality change was covered by the benchmarks but can cover in a follow up PR if we need to?

**Review by SamuelBrand1 — APPROVED** [review]
Like the extra tests.

**Pull request merged**